internal-packages/ai/src/utils/documentContentHelpers.ts

```diff
@@ -0,0 +1,141 @@
+import type { Document } from "../types/documents";
+
+/**
+ * Options for getting document content
+ */
+export interface DocumentContentOptions {
+  /** Whether to include markdownPrepend. Default: true */
+  includePrepend?: boolean;
+  /** Whether to generate prepend if not stored. Default: true */
+  generateIfMissing?: boolean;
+}
+
+/**
+ * Result of getting document content with metadata
+ */
+export interface DocumentContentResult {
+  /** The full content (with or without prepend based on options) */
+  content: string;
+  /** The prepend that was used (if any) */
+  prepend: string | null;
+  /** Number of lines in the prepend */
+  prependLineCount: number;
+  /** Character length of the prepend */
+  prependCharCount: number;
+  /** Whether prepend was generated vs stored */
+  prependWasGenerated: boolean;
+}
+
+/**
+ * Generate markdown prepend for document metadata
+ */
+function generateMarkdownPrepend(metadata: {
```

Comment:
Maybe this should go in /internal-packages/db or similar? It's kind of a domain model function that I expect the /app/ and /db/ to want.